src/document/code.rs

```diff
@@ -196,6 +196,10 @@ impl Line {
             .and_then(|s| s.block_labels.as_ref())
             .map(|l| &l.block_next[..])
             .unwrap_or("");
+        let block_name_sep = settings
+            .and_then(|s| s.block_labels.as_ref())
```

Comment:
I think
```rust
.and_then(|s| s.block_labels.as_ref())
.map(|l| &l.name_separator[..])
.unwrap_or("");
```
could be replaced by
```
.and_then(|s| s.block_labels.as_deref())
.unwrap_or_default();
```

(Generally speaking, I would prefer `foo.as_str()` over `&foo[..]` as the intent is easier to read.)

Reply:
So you mean constructing an entire default `BlockLabels` struct?

Reply:
> So you mean constructing an entire default BlockLabels struct?


No, I just meant replacing `.map(|l| &l.name_separator[..])` by `.as_deref()` to get from `&Option<String>` to `Option<&str>` and replacing `.unwrap_or("")` by the more speaking `unwrap_or_default()`.

Reply:
`name_sep` is gone now, but it still applies to `comment_end`